orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py

```diff
@@ -0,0 +1,654 @@
+            self.bias = Parameter(torch.empty(
+                output_size,
+                device=torch.cuda.current_device(),
+                dtype=torch.float))
+
+            # Always initialize bias to zero.
+            with torch.no_grad():
+                self.bias.uniform_()
+
+        def forward(self, model_input):
+            x = model_input + self.bias
+            y1 = self.inplace_op(self.bias, x)
+            y2 = x.add(self.bias)
+            out = y1 + y2
+            return out
+
+    output_size = 1024
+
+    def model_builder():
+        return InplaceUpdateInputAsOutputRequireGradWithMarkDirtyModel(output_size)
+
+    def input_generator():
+        return torch.randn(output_size, dtype=torch.float)
+
+    # generate a label that have same shape as forward output.
+    label_input = torch.ones([output_size])
+
+    run_training_test_and_compare(model_builder, input_generator, label_input)
+
+
+def test_EvalTest():
+    class EvalTestFunction(torch.autograd.Function):
+        @staticmethod
+        # bias is an optional argument
+        def forward(ctx, x):
+            ctx.save_for_backward(x)
+            return x * 0.5 * (1.0 + torch.tanh(0.79788456 * x * (1 + 0.044715 * x * x)))
+
+        @staticmethod
+        def backward(ctx, grad_output):
+            x = ctx.saved_tensors
+            return None
+
+    class EvalTestModel(torch.nn.Module):
+        def __init__(self, output_size):
+            super(EvalTestModel, self).__init__()
+            self.custom_fn = EvalTestFunction.apply
+            self.bias = Parameter(torch.empty(
+                output_size,
+                device=torch.cuda.current_device(),
+                dtype=torch.float))
+
+            # Always initialize bias to zero.
+            with torch.no_grad():
+                self.bias.uniform_()
+
+        def forward(self, model_input):
+            # model_input did not require_grad
+            out = self.custom_fn(model_input)
+            return out + self.bias
+
+    output_size = 1024
+
+    def model_builder():
+        return EvalTestModel(output_size)
+
+    def input_generator():
+        return torch.randn(output_size, dtype=torch.float)
+
+    # generate a label that have same shape as forward output.
+    label_input = torch.ones([output_size])
+
+    # Test pure inferencing scenarios, when inputs don't requires_grad.
+    run_evaluate_test_and_compare(model_builder, input_generator, label_input)
+
+
+def test_TwoOutputFunction():
+    class TwoOutputFunction(torch.autograd.Function):
+        @staticmethod
+        # bias is an optional argument
+        def forward(ctx, x, y):
+            ctx.save_for_backward(x, y)
+            w = x + y
+            z = x * y
+            return w, z
+
+        @staticmethod
+        def backward(ctx, dw, dz):
+            x, y = ctx.saved_tensors
+            # dL/dx = dL/dw * dw/dx + dL/dz * dz/dx
+            # dw/dx = 1
+            # dz/dx = y
```

Comment:
May we remove dead code/comments from this file?

Reply:
They are comments to the following code. I will make it clearer. Thanks.